web/src/components/DateRangeDropdown.tsx

```diff
@@ -0,0 +1,111 @@
+import * as React from "react";
+import { addMinutes } from "date-fns";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/src/components/ui/select";
+import { type DashboardDateRange } from "@/src/pages/project/[projectId]";
+
+import {
+  DEFAULT_AGGREGATION_SELECTION,
+  isValidOption,
+  type DashboardDateRangeAggregationOption,
+  type TableDateRangeAggregationOption,
+  tableDateRangeAggregationSettings,
+  dashboardDateRangeAggregationSettings,
+  DASHBOARD_AGGREGATION_PLACEHOLDER,
+  type DateRangeOptions,
+} from "@/src/utils/date-range-utils";
+
+type DateRangeDropdownProps = {
+  type: "dashboard" | "table";
+  selectedOption: DateRangeOptions;
+  setDateRangeAndOption: (
+    option: DateRangeOptions,
+    date?: DashboardDateRange,
+  ) => void;
+};
+
+const DateRangeDropdown: React.FC<DateRangeDropdownProps> = ({
+  type,
+  selectedOption,
+  setDateRangeAndOption,
+}) => {
+  const onDropDownSelection = (value: DateRangeOptions) => {
+    if (isValidOption(value)) {
+      let fromDate: Date;
+      if (type === "dashboard") {
+        const setting =
+          dashboardDateRangeAggregationSettings[
+            value as DashboardDateRangeAggregationOption
+          ];
+        fromDate = addMinutes(new Date(), -setting.minutes);
+      } else if (type === "table") {
+        if (value.toString() === DEFAULT_AGGREGATION_SELECTION) {
+          setDateRangeAndOption(DEFAULT_AGGREGATION_SELECTION, undefined);
+          return;
+        }
+        const setting =
+          tableDateRangeAggregationSettings[
+            value as TableDateRangeAggregationOption
+          ];
+        fromDate = addMinutes(new Date(), -setting.minutes);
+      } else {
+        setDateRangeAndOption(DEFAULT_AGGREGATION_SELECTION, undefined);
+        return;
+      }
+      setDateRangeAndOption(value, {
+        from: fromDate,
+        to: new Date(),
```

Comment:
do we need to set a to? it feels like it would be nice to keep this optional as a filter that is only used on dashboard if user sets explicit range